File: runHouses.R
#!/Program Files/R/R-4.3.1/bin/x64/Rgui Rscript

#number of runs
N = 100 

# start keeping track of time
start_time <- Sys.time()

# create and access output file
sink("housesOutputR.txt")

#read input csv and print summary stats to the output file - repeated 100 times
for (i in 1:N) {
    houses = read.csv(file = "housesInput.csv", header = TRUE)
    print(summary(houses)) 
}

#leave the output file
sink()

#calculate execution time for all runs
end_time <- Sys.time()
runtime <- end_time - start_time

cat("Execution Time (sec) for 100 runs = ", runtime)
cat("\nAverage Execution Time (sec) per run = ", runtime/100)

File: runHouses.py
#!/usr/bin/env python


#####!/Users/Moria/AppData/Local/Programs/Python/Python310/ python

#####!/usr/bin/env python

import pandas as pd
import time

# number of runs
N = 100 

# start keeping track of time
start = time.process_time()

#read input file, calculate and write summary stats to output file - repeat 100 times
with open('housesOutputPy.txt', 'wt') as outfile:
    for i in range(N):
        houses = pd.read_csv("housesInput.csv")
        outfile.write(houses.describe().to_string(header=True, index=True))
        outfile.write("\n")

#calculate execution time
total = time.process_time() - start

print("Execution Time (sec) for 100 runs = ", total)
print("Average Execution Time (sec) per run = ", total/100)
